cegui/src/text/LegacyTextParser.cpp

```diff
@@ -106,16 +113,7 @@ bool LegacyTextParser::parse(const String& inText, std::u32string& outText,
     //???TODO: use stack struct for tag params to make the parser itself stateless?
 
     // Initialize formatting parameters with default values
-    d_colours = Colour(0xFFFFFFFF);
-    d_bgColours = Colour(0x00000000);
-    d_font = nullptr;
-    d_padding = Rectf(0.f, 0.f, 0.f, 0.f);
-    d_imageSize = Sizef(0.f, 0.f);
-    d_vertFormatting = VerticalImageFormatting::BottomAligned;
-    d_outlineSize = 0.f;
-    d_underline = false;
-    d_strikeout = false;
-    d_styleChanged = true;
+    handleReset("full");
```

Comment:
I don't like using interface string constants in the internal logic. Consider making a method `resetStyle(bool color, bool font ...)` or `resetColor`, `resetFont` etc, and calling them here and in `handleReset`. String constants should only appear in `handleReset` where they are parsed into internal commands.

Reply:
fixed